pkg/cloudprovider/vm.go

```diff
@@ -43,7 +41,6 @@ func NewVirtualMachinesClient(config config.AzureConfig, spt *adal.ServicePrinci
 
 	reporter, err := metrics.NewReporter()
 	if err != nil {
```

Comment:
Are the error here identifiable enough ? Should we wrap up the error to indicate where the issue was ?